src/main/java/org/jabref/gui/preferences/TableColumnsTab.java

```diff
@@ -96,81 +96,80 @@
                             t.getTablePosition().getRow()).setName(t.getNewValue());
-                });
-        column.setCellValueFactory(new PropertyValueFactory<>("length"));
-        column.setOnEditCommit(
-                (TableColumn.CellEditEvent<TableRow, Double> t) -> {
-                    t.getTableView().getItems().get(
-                            t.getTablePosition().getRow()).setLength(t.getNewValue());
+                    // Since data is an ObservableList, updating it updates the displayed field name.
+                    this.data.set(t.getTablePosition().getRow(), new TableRow(t.getNewValue()));
+                    // Update the User Preference of COLUMN_NAMES
+                    List<String> tempColumnNames = this.prefs.getStringList(this.prefs.COLUMN_NAMES);
+                    tempColumnNames.set(t.getTablePosition().getRow(), t.getNewValue());
+                    this.prefs.putStringList(this.prefs.COLUMN_NAMES,tempColumnNames);
                 });
 
+        tableRows.clear();
+        tableRows.addAll(data);
         colSetup.setItems(data);
-        colSetup.getColumns().addAll(field,column);
+        colSetup.getColumns().add(field);
+
         final TextField addName = new TextField();
         addName.setPromptText("name");
         addName.setMaxWidth(field.getPrefWidth());
         addName.setPrefHeight(30);
-        final TextField addLast = new TextField();
-        addLast.setMaxWidth(column.getPrefWidth());
-        addLast.setPromptText("width");
-        addLast.setPrefHeight(30);
         BorderPane tabPanel = new BorderPane();
         ScrollPane sp = new ScrollPane();
         sp.setContent(colSetup);
         tabPanel.setCenter(sp);
 
         HBox toolBar = new HBox();
-        Button addRow = new Button("Add");
+        addRow = new Button("Add");
         addRow.setPrefSize(80, 20);
-        addRow.setOnAction( e -> {
-            if (!addLast.getText().isEmpty()) {
-                TableRow tableRow = addLast.getText().matches("[1-9][0-9]") ? new TableRow(addName.getText(), Integer.valueOf(addLast.getText())) : new TableRow(addName.getText());
-                addName.clear();
-                addLast.clear();
-                data.add(tableRow);
+        addRow.setOnAction(e -> {
+            TableRow tableRow = new TableRow(addName.getText());
+            addName.clear();
+            data.add(tableRow);
+            tableRows.clear();
+            tableRows.addAll(data);
+            colSetup.setItems(data);
+            tableChanged = true;
+            colSetup.refresh();
+
+        });
+
+        deleteRow = new Button("Delete");
+        deleteRow.setPrefSize(80, 20);
+        deleteRow.setOnAction(e -> {
+            if (colSetup.getFocusModel() != null && colSetup.getFocusModel().getFocusedIndex() != -1) {
+                tableChanged = true;
+                int row = colSetup.getFocusModel().getFocusedIndex();
+                TableRow tableRow = data.get(row);
```

Comment:
Here it should also be sufficient to just change `data` without any further updates to `colSetup`.